### PR TITLE
Refactor docs and deployment

### LINKED ISSUE #39: .gitignore missed charts directory
## Describe the Bug

I am 99% certain that `helm-charts/vmpooler/charts/` should have been ignored but is currently included here. We need to exclude that dir and removed the tarball it currently contains from the repo.

### PR DESCRIPTION
**Pull request by yachub**
I think this will simplify the developer experience by adding some clear instructions on how to get started and reduce the number of dockerfiles and gemfiles that need dependency updates. I don't foresee the packages linked to this repository ever hosting anything other than a "prod-all-providers" docker image, so I think we can drop that naming scheme and simply stick with a semantic version for the docker image (aka "appVersion"), and the tags/release used for the repository can be the helm chart version instead. You can easily find what docker tag is used in a particular chart version by just checking the history of `Chart.yaml`.

Though the release workflow does need to use a bot token to commit to the default branch, it should be safe since we have the [require approval for all outside contributors](https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks) enabled. Received approval from security that it should be good to go.

The release workflow will skip the docker build step if a container tag already exists matching the "appVersion", which is needed in case the docker image hasn't changed, but the chart has.

As part of the versioning change, tag the latest commit on `main` with `1.19.0`.

Also fixes #39

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,11 +1,24 @@
...
+  - [Contributing](#contributing)
+  - [License](#license)
+
 This repo contains Dockerfiles and a Helm chart that can be used to deploy [VMPooler](https://github.com/puppetlabs/vmpooler). The Release Engineering team at Puppet uses the code here as part of operating our VMPooler instances.
 
 ## Docker Registry
 
 The GitHub Actions in this repository publish images to GitHub Packages. You can browse the VMPooler containers [here](https://github.com/puppetlabs/vmpooler-deployment/pkgs/container/vmpooler-deployment%2Fvmpooler).
 
+The vmpooler docker image is tagged using
```
Comment (isaac-hammes) [inline README.md:20]: I think you meant to say more here. Just wanted to let you know if it was an accident.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        with:
+          args: >-
+            --future-release ${{ steps.nv.outputs.version }}
+        env:
+          CHANGELOG_GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Validate Changelog
+        run : |
+          set -e
+          if output=$(git status --porcelain) && [ ! -z "$output" ]; then
```
Comment (e-gris) [inline .github/workflows/release.yml:55]: Trivial: this can be written more concisely:

    if [[ -n $(git status --porcelain) ]]; then

**Review by e-gris — COMMENTED** [review]
Spotted one bit of trivia, more in-depth look to come.

**Review by e-gris — COMMENTED** [review]
Another trivial style comment: remove the `.sh` extension on executable sh/bash files. This is a common antipattern of encoding implementation into the interface.

File: README.md [hunk 3]
```diff
@@ -21,14 +34,68 @@ $ helm repo add vmpooler-deployment https://puppetlabs.github.io/vmpooler-deploy
...
+- [Docker Compose](https://docs.docker.com/compose/install/)
+
+1. Become familiar with the configuration file `docker/vmpooler.yaml` as described in [VMPooler](https://github.com/puppetlabs/vmpooler) Core.
+   - The default configuration file only enables the dummy provider built into [VMPooler](https://github.com/puppetlabs/vmpooler) Core. See each project below for documentation on how to use provider specific options:
+      - [vmpooler-provider-ec2](https://github.com/puppetlabs/vmpooler-provider-ec2)
+      - [vmpooler-provider-gce](https://github.com/puppetlabs/vmpooler-provider-gce)
+      - [vmpooler-provider-vsphere](https://github.com/puppetlabs/vmpooler-provider-vsphere)
+2. Chose a Development method:
+   - Develop via local source:
+      1. Clone all of the known vmpooler projects under a common directory, for example:
```
Comment (e-gris) [inline README.md:53]: How do I find the list of `all the known vmpooler projects`? Can we have a script that does this?

```$ clone-known-vmpooler-projects```

File: README.md [hunk 4]
```diff
@@ -21,14 +34,68 @@ $ helm repo add vmpooler-deployment https://puppetlabs.github.io/vmpooler-deploy
...
+
+           ```bash
+           |-- vmpooler-projects
+           |  |--vmpooler-deployment
+           |  |--vmpooler-provider-ec2
+           |  |--vmpooler-provider-gce
+           |  |--vmpooler-provider-vsphere
+           ```
+
+      2. From the `vmpooler-projects` directory run `docker build -t vmpooler-local -f vmpooler-deployment/docker/Dockerfile.local .`
```
Comment (e-gris) [inline README.md:63]:
```
[com:~/testdir/vmpooler-projects]$ docker build -t vmpooler-local -f vmpooler-deployment/docker/Dockerfile.local .
ERROR: (gcloud.auth.docker-helper) There was a problem refreshing your current auth tokens: ('invalid_grant: Bad Request', {'error': 'invalid_grant', 'error_description': 'Bad Request'})
Please run:

  $ gcloud auth login

to obtain new credentials.

If you have already logged in with a different account:

    $ gcloud config set account ACCOUNT

to select an already authenticated account to use.
```

Comment (e-gris) [reply README.md:63]: After running `gcloud auth login`:

```
[com:~/testdir/vmpooler-projects]$ docker build -t vmpooler-local -f vmpooler-deployment/docker/Dockerfile.local .
Sending build context to Docker daemon  5.221MB
Step 1/16 : FROM jruby:9.3.6-jdk
 ---> 706cb98073fe
Step 2/16 : LABEL org.opencontainers.image.description "VMPooler plus all known providers"
 ---> Using cache
 ---> fe341449d1f1
Step 3/16 : ENV DEBIAN_FRONTEND=noninteractive
 ---> Using cache
 ---> 4cd0776c040b
Step 4/16 : ENV LOGFILE=/dev/stdout
 ---> Using cache
 ---> 69e5f71b6d56
Step 5/16 : ENV RACK_ENV=production
 ---> Using cache
 ---> 60669f477612
Step 6/16 : RUN apt-get update -qq &&     apt-get upgrade -y &&     apt-get install -y --no-install-recommends make git netbase &&     apt-get clean autoclean &&     apt-get autoremove -y &&     rm -rf /var/lib/apt/lists/*
 ---> Using cache
 ---> 2e25da848515
Step 7/16 : COPY vmpooler /opt/vmpooler
COPY failed: file not found in build context or excluded by .dockerignore: stat vmpooler: file does not exist
```

**Review by e-gris — COMMENTED** [review]
The overall redesign here is well-done

I can continue with a QA-ish approach but currently blocked on the `docker build` issue.

File: README.md (hunk 1 above)
Comment (yachub) [reply README.md:20]: Ah thanks, definitely saw a squirrel or something in the middle of that sentence.

File: README.md (hunk 3 above)
Comment (yachub) [reply README.md:53]: Oh darn, thought that I had listed those in the Readme, I'll add them.

File: README.md (hunk 4 above)
Comment (yachub) [reply README.md:63]: Ah yes, the build context needs to be in the `docker` directory. I'll specify that.

I think the gcloud prompt is unrelated though, there's nothing gcloud specific in there. I guess their helper does that when access expires/is invalid

File: .github/workflows/release.yml (hunk 2 above)
Comment (yachub) [reply .github/workflows/release.yml:55]: Hm, I don't know how I originally ended up with that, but yes thanks that is much better.

**Review by e-gris — APPROVED** [review]

**Pull request merged**